src/util/i18n/i18n.ts

```diff
@@ -95,3 +95,25 @@ export const localizedNumberParser = (
 
   return (formattedString: string): number => pipe(formattedString, transformSeparators, parseFloat)
 }
+
+export type LimitedDateTimeFormatOptions = Pick<
```

Comment:
Why the notion of `limited`?
You are creating your own utility function with some input type, user has no need to know if you are using, for the definition, some restricted types from another context like `Intl`.
```suggestion
export type DateTimeFormatOptions = Pick<
```

Reply:
The idea was to hint that only some date format options can be provided.
The naming is probably not the most suited to express that idea.
Wanted to provide a different name that `DateTimeFormatOptions` type than `Intl` option type to avoid confusions.
May be `PartialDateTimeFormatOptions` would have been most appropriate?

Also, this function is finally just a factory that creates an Intl instance.
It might be sufficient to only use `formatDateToPattern` as utility.

What do you think?

Reply:
@0xLodz 
You pick some properties from another type to declare your own type, IMHO, that is not limited.
It may be a matter of perspective, this is why I just added my thoughts about that.
Feel free to adjust the naming or keep like that, it must make sense for you 😌